File: VB/Helper/RowHeaderHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.XtraGrid.Views.BandedGrid
Imports DevExpress.Utils.Drawing
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Drawing

Namespace WindowsApplication1
	Public Class RowHeaderHelper

		Private _View As AdvBandedGridView
		Private _Column As New BandedGridColumn()

		Private Sub InitializeColumn(ByVal bandCaption As String, ByVal caption As String, ByVal fixedStyle As FixedStyle)
			Dim band As GridBand = _View.Bands.AddBand(bandCaption)
			band.Fixed = fixedStyle
			_Column.FieldName = "Unbound" & _View.Columns.Count
			_Column.Caption = caption
			_Column.UnboundType = DevExpress.Data.UnboundColumnType.String
			_Column.Visible = True
			_Column.OptionsColumn.AllowFocus = False
			_Column.OptionsFilter.AllowFilter = False
			_Column.OptionsColumn.AllowGroup = DevExpress.Utils.DefaultBoolean.False
			_Column.OptionsColumn.AllowSort = DevExpress.Utils.DefaultBoolean.False
			band.Columns.Add(_Column)
		End Sub

		Public Sub New(ByVal view As AdvBandedGridView, ByVal bandCaption As String, ByVal caption As String, ByVal fixedStyle As FixedStyle)
			_View = view
			InitializeColumn(bandCaption, caption, fixedStyle)
			AddHandler _View.CustomDrawCell, AddressOf _View_CustomDrawCell
		End Sub

		Private Sub _View_CustomDrawCell(ByVal sender As Object, ByVal e As RowCellCustomDrawEventArgs)
			If e.Column IsNot _Column Then
				Return
			End If
			MyArgs.DrawHeader(e.Cache, e.Column, GetColumnCaption(e.RowHandle), e.Bounds)

		End Sub

		Private Function GetColumnCaption(ByVal rowHandle As Integer) As String
			Return String.Format("Row {0}", rowHandle + 2)
		End Function
	End Class
End Namespace

File: CS/WindowsApplication1/Helper/RowHeaderHelper.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Columns;
using DevExpress.XtraGrid.Views.BandedGrid;
using DevExpress.Utils.Drawing;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Drawing;

namespace WindowsApplication1
{
    public class RowHeaderHelper
    {

        private AdvBandedGridView _View;
        BandedGridColumn _Column = new BandedGridColumn();

        private void InitializeColumn(string bandCaption, string caption, FixedStyle fixedStyle)
        {
            GridBand band = _View.Bands.AddBand(bandCaption);
            band.Fixed = fixedStyle;
            _Column.FieldName = "Unbound" + _View.Columns.Count;
            _Column.Caption = caption;
            _Column.UnboundType = DevExpress.Data.UnboundColumnType.String;
            _Column.Visible = true;
            _Column.OptionsColumn.AllowFocus = false;
            _Column.OptionsFilter.AllowFilter = false;
            _Column.OptionsColumn.AllowGroup = DevExpress.Utils.DefaultBoolean.False;
            _Column.OptionsColumn.AllowSort = DevExpress.Utils.DefaultBoolean.False;
            band.Columns.Add(_Column);
        }

        public RowHeaderHelper(AdvBandedGridView view, string bandCaption, string caption, FixedStyle fixedStyle)
        {
            _View = view;
            InitializeColumn(bandCaption, caption, fixedStyle);
            _View.CustomDrawCell += _View_CustomDrawCell;
        }

        void _View_CustomDrawCell(object sender, RowCellCustomDrawEventArgs e)
        {
            if (e.Column != _Column)
                return;
            MyArgs.DrawHeader(e.Cache, e.Column, GetColumnCaption(e.RowHandle), e.Bounds);

        }

        private string GetColumnCaption(int rowHandle)
        {
            return String.Format("Row {0}", rowHandle + 2);
        }
    }
}
